nattable.rb

```diff
@@ -182,11 +186,13 @@ def remote_key_from_packet(packet)
     src_addr = packet.src_addr
     addr_size = src_addr.size
 
-    b = IO::Buffer.new(addr_size + 4)
-    b.copy(src_addr)
-    b.copy(packet.l4.tuple, addr_size)
+    key = ZERO_STR.byteslice(0, addr_size + 4)
+    IO::Buffer.for(key) do |b|
```

Comment:
It's neat idea, I didn't think about it. Maybe we can do the similar:

```
IO::Buffer.string(size) do |buffer|
  # ...
end # => string
```

Reply:
That interface SGTM. I'm all in to optimizations that reduce the conversion cost bet. the two types.

Reply:
That seems a nice addition, and it's easy to support without extra copying on TruffleRuby, unlike the current usage here (would need one copy to go from Rope to byte[] internally).